client/signup/main.jsx

```diff
@@ -92,12 +95,32 @@ const Signup = React.createClass( {
 		}
 	},
 
+	submitQueryDependencies() {
+		if ( 'undefined' === typeof this.props.queryObject ) {
+			return;
+		}
+		const vertical = this.props.queryObject.vertical;
```

Comment:
This can be turned into a conditional block: 
```
if ( this.props.queryObject.vertical ) {
    const vertical = this.props.queryObject.vertical;
    ... other logic here related to submitting the verticals step ...
}
```

This will allow for better readability when multiple steps get added here. That is if we don't find a better way to handle pre-filling steps with data.